MTA pipeline

zuzulaz opened the pull request.

zuzulaz committed:
Start of MTA pipeline

Start of the pipeline, general structure, Location_data.

zuzulaz committed:
Updates pipelin

Add functions for individual data, location data,  capture data.

LiamDBailey commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+    janitor::clean_names(case = "upper_camel") %>%
+    janitor::remove_empty(which = "rows") %>%
+    #### Convert to corresponding format and rename
+    dplyr::mutate(PopID = "MTA",
```
See comment above about PopID

LiamDBailey commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+
+
+#### --------------------------------------------------------------------------~
+#### QUESTIONS DATA OWNERS
```
Antica's comment from the SPI-Birds spreadsheet:

"their data is in hungarian, but my understanding is that he has sent a translated version.... check before working on this"

Should check with data owner when you contact them.

zuzulaz replied:
Yes, I have seen that Antica's comment. The only word in Hungarian I found is the "gyurutlen" which based on dictionary refers to unringed bird. Appears only several times, majority of unringed individuals are labeled "ringless" or "unringed" in English.

LiamDBailey commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+    #### Remove unringed individuals
+    dplyr::filter(.data$IndvID != "unringed") %>%
+    #### Create capture date
+    dplyr::mutate(CaptureDate = as.Date(paste0(BreedingSeason, "-06-01"))) %>%
```
Check with data owners

LiamDBailey commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+
+  Capture_data <-
+    data %>%
+    dplyr::select(.data$Species, .data$PopID, .data$BreedingSeason,
```
We still want to calculate species from the `species_code` table in case we want to change the codes in the future.

zuzulaz replied:
In this data there is column called Species with "PARMAJ" value. Should I change it to extract the species from the `species_code `instead?

LiamDBailey replied:
Yes, that would be the best approach. In the future we might change the codes (especially if we start getting many different species from overseas), so we want to have it linked to the EURING code.

LiamDBailey commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+#'
+#' \strong{xxx}:
+#'
+#' \strong{Latitude, Longitude} The exact coordinates of the nestboxes are not available.
+#' Data owner provided an map of the location, which can be georeferenced in the case of
+#' interest or necessity of the data user. Currently, only general coordinates
+#' for the location are provided.
```
Is this still true?

zuzulaz replied:
Nope, my mistake. The true is that we do not have coordinates for nestboxes, neither we do have any map. I will ask data owners about that.

LiamDBailey reviewed (changes requested):
A few important things:

- Looks like the MTA data includes 5 different populations (sites), so the PopID should vary between records.
-  Make sure we check that this is the raw data/they can recreate this data. Just like with KIL it looks like they've converted directly to our standard format.

zuzulaz commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+#' A pipeline to produce the standard format for bird study population
+#' at the MTA-PE Evolutionary Ecology Group, Hungary, administered by
+#' Gabor Seres & András Liker.
+#' The data include 5 sites: Veszprém, Balatonfüred, Szentgál, Vilma-puszta and Gulya-domb.
```
Ok. Should I therefore remove the MTA population?

LiamDBailey replied:
Exactly. MTA can be kept as an owner, but is no longer a valid PopID

zuzulaz replied:
Done.

zuzulaz commented on R/format_MTA.R:
```diff
@@ -0,0 +1,371 @@
+#'
+#' This section provides details on data management choices that are unique to
+#' this data. For a general description of the standard format please see see
+#'\href{https://github.com/SPI-Birds/documentation/blob/master/standard_protocol/SPI_Birds_Protocol_v1.0.0.pdf}{here}.
```
👍

zuzulaz committed:
Update after Liam's review

Several changes, add .data$ to the code

zuzulaz committed:
Update, tests

Update few details in the pipeline. Start creating tests, Location data tests finished, others in the process.

zuzulaz committed:
Finished tests

Finished all tests for the pipeline. Small updates to the format function, added description.

zuzulaz committed:
Update to individual data

Add individual "A000" to "unringed" individuals.

zuzulaz committed:
Add populations

Add populations ID to pop_codes and to setup-xyz.R for testing.

zuzulaz committed:
Update species

Change the way of species assignation.

LiamDBailey committed:
Fix global variable note in R CMD Check :warning:

LiamDBailey committed:
Update pop_species_combos :scroll:

LiamDBailey committed:
Merge in master

LiamDBailey committed:
Fix typo from merge :no_entry:

LiamDBailey committed:
Update docs :scroll:

LiamDBailey committed:
Pass R CMD Check :warning:

LiamDBailey commented:
@zuzulaz Made a few minor changes, but otherwise seems fine. I will leave this PR open for when we get the new data, but that will be done by @cwtyson.

zuzulaz commented:
Ok, great, thanks. It means that I can mark this pipeline as finished for me? :)
